src/enigma/config.js

```diff
@@ -1,14 +1,19 @@
-import schema from 'enigma.js/schemas/12.20.0.json';
+import schema from 'enigma.js/schemas/12.170.2.json';
 
 import listCache from './list-cache';
 import layoutCache from './layout-cache';
 
+const searchParams = new URLSearchParams(document.location.search);
+const hostname = searchParams.get('hostname') || 'localhost';
+const port = searchParams.get('port') || '9076';
+const app = searchParams.get('app') || 'enginedata';
+
 const ERR_ABORTED = 15;
-const url = process.env.ENGINE_URL;
 
 const config = {
   schema,
-  createSocket: () => new WebSocket(url || `ws://${location.hostname}:9076/app/${+new Date()}`), // eslint-disable-line
+  app,
```

Comment:
enigma.js does not have an `app` config property, it has a `url` one though. So in the suggestion above, the `?session=` value would be `url` here.